Move `@types` packages to `dependencies` from `devDependencies`

# Bug report

* [x] I have checked other issues to make sure this is not a duplicate.

## Describe the bug

I have a following error after running `tsc --noEmit`:
```
error TS7016: Could not find a declaration file for module 'yup'
```
This should be fixed by moving all `@types` modules to dependencies, not devDependencies (I should not install type definitions for package that is not my dependency.

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
     "postinstall": "lightercollective postinstall"
   },
   "dependencies": {
+    "@types/yup": "0.26.9",
     "lightercollective": "^0.2.0",
     "object-hash": "^1.3.1",
     "yup": "^0.27.0"
@@ -29,7 +30,6 @@
     "@types/node": "10.12.26",
     "@types/object-hash": "1.2.0",
     "@types/request-promise-native": "1.0.15",
-    "@types/yup": "0.26.9",
     "apollo-server": "2.4.2",
     "codecov": "3.2.0",
     "coveralls": "3.0.2",

